tests/integration/helpers.py

```diff
@@ -0,0 +1,146 @@
+from pathlib import Path
+
+import yaml
+from pytest_operator.plugin import OpsTest
+from workload import Prometheus
+
+
+async def unit_address(ops_test: OpsTest, app_name: str, unit_num: int) -> str:
+    """Find unit address for any application.
+
+    Args:
+        ops_test: pytest-operator plugin
+        app_name: string name of application
+        unit_num: integer number of a juju unit
+
+    Returns:
+        unit address as a string
+    """
+    status = await ops_test.model.get_status()
+    return status["applications"][app_name]["units"][f"{app_name}/{unit_num}"]["address"]
+
+
+async def check_prometheus_is_ready(ops_test: OpsTest, app_name: str, unit_num: int) -> bool:
+    """Check if Prometheus server responds to HTTP API requests.
+
+    Args:
+        ops_test: pytest-operator plugin
+        app_name: string name of Prometheus application
+        unit_num: integer number of a Prometheus juju unit
+
+    Returns:
+        True if Prometheus is responsive else False
+    """
+    host = await unit_address(ops_test, app_name, unit_num)
+    prometheus = Prometheus(host=host)
+    is_ready = await prometheus.is_ready()
+    assert is_ready
```

Comment:
In general, I'd say that asserts should be done in the test case, and that this function should return a value for the test case to assert. This usually turns out to be a good practice later, as tests evolve and grow more complex, as no function except the test case itself will fail a test, making it easier to maintain. 

**If** you agree, please create an issue for addressing it later, after your vacation.

Reply:
Created issue with items to be picked up in next PR : https://github.com/canonical/prometheus-operator/issues/156